pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -275,11 +276,19 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
 	// apply components
 	result, err := r.applyComponents(instance)
 	if err != nil {
-		klog.Errorf("failed to deploy components: %v", err)
+		klog.Errorf("failed to deploy performance profile %q components: %v", instance.Name, err)
 		return reconcile.Result{}, err
 	}
 
-	// TODO: we need to update the status
+	if err := r.updateStatus(instance); err != nil {
```

Comment:
We never get here when `applyComponents()` fails a few lines above. I think we want to update the status based on such an error.

Reply:
done